[RuntimePermissions] Modernize watch app

alexvanyo opened the pull request:
Updates the wear portion of the `RuntimePermissions` sample, converting from Java to Kotlin and introducing coroutines, `ViewBinding`, and the `androidx` wear library.

codingjeremy commented on RuntimePermissionsWear/Wearable/src/main/res/drawable/action_background.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Missing open source headers.

codingjeremy commented on RuntimePermissionsWear/Wearable/src/main/res/drawable/ic_send_to_mobile.xml:
```diff
@@ -0,0 +1,11 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Missing open source headers.

codingjeremy commented on RuntimePermissionsWear/Wearable/src/main/res/drawable/ic_very_dissatisfied.xml:
```diff
@@ -0,0 +1,16 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Missing open source headers.

codingjeremy commented on RuntimePermissionsWear/Wearable/src/main/res/drawable/ic_very_satisfied.xml:
```diff
@@ -0,0 +1,16 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Missing open source headers.

codingjeremy commented on RuntimePermissionsWear/Wearable/src/main/res/layout/activity_main.xml:
```diff
@@ -1,5 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!--
+<?xml version="1.0" encoding="utf-8"?><!--
```
Nit, I know studio does this, but the comment should really start on the next line. ^_^;;

codingjeremy replied:
Wasn't fixed but I'll approve since maybe it was missed?

codingjeremy commented on RuntimePermissionsWear/Wearable/src/main/res/values/colors.xml:
```diff
@@ -0,0 +1,6 @@
+    <color name="action_blue">#0086D4</color>
+    <color name="permission_granted">#20B000</color>
+    <color name="permission_denied">#B00020</color>
+</resources>
```
Empty new line needed at the end (or it shows up as a red negative in github.

alexvanyo commented on RuntimePermissionsWear/Wearable/src/main/java/com/example/android/wearable/runtimepermissions/IncomingRequestWearService.kt:
```diff
@@ -0,0 +1,166 @@
+
+    override fun onMessageReceived(messageEvent: MessageEvent) {
+        Log.d(TAG, "onMessageReceived(): $messageEvent")
+        scope.launch {
```
I can add a comment here!

It is a bit confusing that `suspend` gets passed down through a bunch of functions to the place where it's actually necessary, but deferring the `scope.launch` makes it a bit harder to reason about the order of things.

Being able to initially switch to a suspending context implies "the work we need to do based on the message is asynchronous, so now everything we're doing might be asynchronous." I'll write something to that affect here in a comment!

alexvanyo commented on RuntimePermissionsWear/Wearable/src/main/java/com/example/android/wearable/runtimepermissions/RequestPermissionOnPhoneActivity.kt:
```diff
@@ -0,0 +1,58 @@
+        binding = ActivityRequestPermissionOnPhoneBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.openOnPhoneContainer.setOnClickListener {
+            setResult(RESULT_OK)
+            finish()
+        }
```
Yeah, my preference is the other way so I think I'll leave it.
The unused-but-necessary `View` argument bothers me a bit.

codingjeremy replied:
Ah, I see, that makes sense. SGTM.

alexvanyo commented:
@codingjeremy should be ready for another look, I added/updated all open source headers, added new lines and added a comment around the coroutines usage in the `Service`.

alexvanyo commented:
Thanks @codingjeremy !
